subsys/bluetooth/audio/ascs.c

```diff
@@ -814,17 +882,21 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 	}
 
 	for (size_t i = 0; i < ARRAY_SIZE(session->ases); i++) {
-		struct bt_ascs_ase *ase = &session->ases[i];
-		struct bt_audio_stream *stream = ase->ep.stream;
-
-		if (ase->ep.status.state != BT_AUDIO_EP_STATE_IDLE) {
-			/* ase_process will handle the final state transition into idle state */
-			ase_release(ase);
-		}
+		if (session->ases[i]) {
+			struct bt_ascs_ase *ase = session->ases[i];
+				struct bt_audio_stream *stream = ase->ep.stream;
```

Comment:
Would look cleaner if you first declare the variables

```
struct bt_ascs_ase *ase = &session->ases[i];
struct bt_audio_stream *stream;

if (ase == NULL)  {
   continue;
}

stream = ase->ep.stream;
```

Plus, you won't need to change the code below (so that the history is unchanged)